src/boost.beast/session.h

```diff
@@ -0,0 +1,119 @@
+#pragma once
+#include "helper.h"
+#include "request.h"
+
+// Handles an HTTP server connection
+class Session : public std::enable_shared_from_this<Session>
+{
+  public:
+    // Take ownership of the stream
+    explicit Session(tcp::socket &&socket, std::vector<data::person> &person) : m_person(person), m_stream(std::move(socket)), m_lambda(*this) {}
+
+    // Start the asynchronous operation
+    void run()
+    {
+        // We need to be executing within a strand to perform async operations
+        // on the I/O objects in this session. Although not strictly necessary
+        // for single-threaded contexts, this example code is written to be
+        // thread-safe by default.
+        net::dispatch(m_stream.get_executor(), beast::bind_front_handler(&Session::doRead, shared_from_this()));
+    }
+
+    void doRead()
+    {
+        // Make the request empty before reading,
+        // otherwise the operation behavior is undefined.
+        m_req = {};
+
+        // Set the timeout.
+        m_stream.expires_after(std::chrono::seconds(30));
+
+        // Read a request
+        http::async_read(m_stream, m_buffer, m_req, beast::bind_front_handler(&Session::onRead, shared_from_this()));
+    }
+
+    void onRead(beast::error_code ec, std::size_t bytes_transferred)
+    {
+        boost::ignore_unused(bytes_transferred);
+
+        // This means they closed the connection
+        if (ec == http::error::end_of_stream)
+        {
+            return doClose();
+        }
+
+        if (ec)
+        {
+            return fail(ec, "read");
+        }
+
+        // Send the response
+        handleRequest(std::move(m_req), m_lambda, m_person);
+    }
+
+    void onWrite(bool close, beast::error_code ec, std::size_t bytes_transferred)
+    {
+        boost::ignore_unused(bytes_transferred);
+
+        if (ec)
+        {
+            return fail(ec, "write");
+        }
+
+        if (close)
+        {
+            // This means we should close the connection, usually because
+            // the response indicated the "Connection: close" semantic.
+            return doClose();
+        }
+
+        // We're done with the response so delete it
+        m_res = nullptr;
+
+        // Read another request
+        doRead();
+    }
+
+    void doClose()
+    {
+        // Send a TCP shutdown
+        beast::error_code ec;
+        m_stream.socket().shutdown(tcp::socket::shutdown_send, ec);
+
+        // At this point the connection is closed gracefully
+    }
+
+  private:
+    std::vector<data::person> &m_person;
+    // This is the C++11 equivalent of a generic lambda.
```

Comment:
why are we constrained to C++11?

Reply:
we don't, removed this line. I will change this to generic lambda